docs/docs/50-user-guide/60-reference-docs/80-webhook-receivers/generic.md

```diff
@@ -4,8 +4,234 @@ sidebar_label: Generic
+:::
+
+:::info
+"Refreshing" a `Warehouse` resource means enqueuing it for immediate
+reconciliation by the Kargo controller, which will execute the discovery of new
+artifacts from all repositories to which that `Warehouse` subscribes.
+:::
+
+## Configuring the Receiver
+
+A Generic webhook receiver must reference a Kubernetes `Secret` resource with
+a `secret` key in its data map.
+
 :::info
-__Coming soon!__
+Kargo incorporates the secret into the generation of a hard-to-guess URL for the
+receiver. This URL serves as a _de facto_
+[shared secret](https://en.wikipedia.org/wiki/Shared_secret) and authentication
+mechanism.
+:::
+
+:::note
+The following commands are suggested for generating and base64-encoding a
+complex secret:
+
+```shell
+secret=$(openssl rand -base64 48 | tr -d '=+/' | head -c 32)
+echo "Secret: $secret"
+echo "Encoded secret: $(echo -n $secret | base64)"
+```
+
+:::
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: wh-secret
+  namespace: kargo-demo
+  labels:
+    kargo.akuity.io/cred-type: generic
+data:
+  secret: <base64-encoded secret>
+---
+apiVersion: kargo.akuity.io/v1alpha1
+kind: ProjectConfig
+metadata:
+  name: kargo-demo
+  namespace: kargo-demo
+spec:
+  webhookReceivers:
+    - name: my-receiver
+      generic:
+        secretRef:
+          name: wh-secret
+      actions:
+        - action: Refresh
+          # Parameters can optionally be defined for use in expressions
+          parameters:
+            - targetEvent: "push"
+            - headerKey: "X-Event-Type"
+          # Only perform this action if this expression is satisfied
+          matchExpression: "request.header(params.headerKey) == params.targetEvent"
+          targets:
+            # This target is designated via static name
+            - kind: Warehouse
+              name: my-warehouse
+            # This target is designated via expression derived name
+            - kind: Warehouse
+              name: "${{ request.body.repository.name }}"
+            # This target omits the name all together and designates any
+            # resources that fulfill the label and index selector criteria
+            - kind: Warehouse
+              labelSelector:
+                matchLabels:
+                  environment: production
+                matchExpressions:
+                  - key: tier
+                    operator: In
+                    values: ["critical", "high"]
+              indexSelector:
+                matchExpressions:
+                  - key: subscribedURL
+                    operator: In
+                    value: "${{ normalizeGit(request.body.repository.url) }}"
+```
+
+:::note
+`name`, `labelSelector`, and `indexSelector` are all
+optional. However, at least one of them must be specified.
+:::
+
+## Designating Targets
+
+There are 3 different ways of designating `Target`s:
+
+1. [Static](#static)
+1. [Expression-Derived](#expression-derived)
+1. [Selector-Based](#selector-based)
```

Comment:
This is a list of apples and oranges. Static text and expressions are two ways for providing values. Selectors are a mechanism into which you might plug those values.

This list would be better as:

1. By name
2. By labels
3. By values in an index

Any of those three _support_ static values or expressions.